### DISCUSSION TITLE
Optimal parameters (unet, text encoder, concept steps)

### DISCUSSION THREAD

**Discussion by specblades** [TheLastBen/fast-stable-diffusion#981] [Q&A]
Before I have had too many unsuccessful attempts (4 so far), I want to ask - how can I calculate in advance what parameters to set?

I would be insanely happy if you could share your experiences.

For example.
I train for subjects.
Different representations of the same class of objects, 40 images, each named with 4 tags, 200 concept imgs
tags: men/women, armor, leather/chain/heavy/futuristic, helmet/helmetless

As I understand it (for the next attempt, prev was totally garbage), the approximate values should be: 4000 unet, 1200-1600 text encoder, 600-700 encoder concept steps.

Correct me please.

**Comment by TheLastBen** [comment]
don't use tags for the images, use only one single identifier for all the instance images, and for the concept images, you don't need to rename them or crop them, just upload them.

for 40 images and 200 concept images, start with 3500 UNet, 600 text_enc, and 800 concept_text_enc

**Reply by nawnie** [reply]
so in general if it has people or subjects we need about 30% more concept steps over text encoder?

**Reply by TheLastBen** [reply]
yes, since the concept weight won't get channeled into a single token, it's fine to raise the steps a bit, just make sure they're diverse.

**Reply by nawnie** [reply]
well, its a frame from every 2 seconds of ... 4 movies now? should be diverse...

training it on my already made futurama did solve will smith from looking like hermes, so imma try it from the start.

**Reply by specblades** [reply]
> don't use tags for the images, use only one single identifier for all the instance images, and for the concept images, you don't need to rename them or crop them, just upload them.
> 
> for 40 images and 200 concept images, start with 3500 UNet, 600 text_enc, and 800 concept_text_enc

Hmm, okay, I can probably split my dataset into 8 identifiers, should I keep the numbers you provided?
8x5imgs

**Reply by TheLastBen** [reply]
My recommendations are very general, but it really depends on the instance images, so you have to experiment and keep training until you get a good result, the only think that you should avoid doing is overtraining the text encoder.

**Reply by specblades** [reply]
So I think I finally got something and now without the body horror, hehe. The armor looks good, but it feels like the model is doing a pale generation and can't quite pick up the artists styles. Everything is a little blurry and not contrasty.

Can you give me some advice on what might be the problem? Should I add a little more text encoder steps?

![image](https://user-images.githubusercontent.com/113993746/207127047-42a00e8b-011f-46ad-99de-2e94a2e28e1c.png)
![image](https://user-images.githubusercontent.com/113993746/207127516-8ef3a29d-b695-4e3f-a647-9421a3650c2d.png)

**Reply by TheLastBen** [reply]
reduce the number of instance images to 20 for a stronger style presence

**Reply by specblades** [reply]
This time I trained a model with these parameters: 32 images, 4 tokens of 8 images, 3000 steps, 480 text encoder, 580 concept encoder. Tell me if I set the parameters wrong.
Next time I will try 20 images, 2 tokens of 10. But in this case, I seem to lose the ability to point generate the right type of armor. I dunno

**Reply by specblades** [reply]
Oh, I think I'm beginning to understand, variety is achieved through training the concept encoder, and the instance images should not be a lot

**Reply by TheLastBen** [reply]
exactly, you let the concept images help with the diversity, and the instance images with the visual style. 
the concept images can be as small as 256x256, you can upload up to 1000, make sure they fall in the same category as your subject for example "warrior", "knight" ... etc

**Reply by SAC020** [reply]
> don't use tags for the images

why not use tags / caption the images?

**Reply by TheLastBen** [reply]
it will cause overfitting
